### PR TITLE
Track signed in event on sign up

### LINKED ISSUE #10230: Analytics: Bump the the signed_in event at the end of account creation
For consistency between platforms it would be nice if we could bump the `signed_in` event at the end of the account creation flow once the user's auth token has been acquired.

### PR DESCRIPTION
**Pull request by planarvoid**
Fixes #10230

I've tried to identify the places where we're sure we have a token during the signup. I think there are 2 things that could happen. One is a successful Google signup and the other one is going through the magic link and getting the token there. I've covered these two use cases but there might be something wrong because the flow is really hard to follow. 

As part of this PR I did a very small refactoring where I moved the `AccountStore` and `SiteStore` to the constructor of the `LoginAnalyticsModule` instead of passing it as parameters. I think that's the correct approach since the module is injected.

Let me know if this approach makes sense!

To test:
* Check that the sign-up flow isn't broken.

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginActivity.java [hunk 1]
```diff
@@ -779,7 +779,7 @@ public void onGoogleLoginFinished() {
     @Override
     public void onGoogleSignupFinished(String name, String email, String photoUrl, String username) {
         AnalyticsTracker.track(AnalyticsTracker.Stat.SIGNUP_SOCIAL_SUCCESS);
-
+        mLoginAnalyticsListener.trackAnalyticsSignIn(false);
```
Comment (aforcier) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginActivity.java]: Shouldn't this be `trackAnalyticsSignIn(true)`? I believe the flag is meant to identify self-hosted vs WP.com logins, and WP.com signups should count as the latter.

Comment (aforcier) [reply WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginActivity.java]: (The same would apply to the change in `WPMainActivity`.)

**Comment by aforcier** [comment]
@planarvoid aside from my comment this looks good to me. Definitely 👍 on the DI refactor.

It's probably worth having someone who worked on signup take a look as well though as I agree the flow is fairly complex. @theck13 could you give this a once-over?

**Comment by aerych** [comment]
> can you confirm that's what is intended with the issue?

Basically, once we've obtained the user's bearer token, we want to bump the signed in stat. If that happens immediately after we bump account created, then yes.

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginActivity.java (hunk 1 above)
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginActivity.java]: you're right of course! good catch

**Comment by planarvoid** [comment]
@theck13 You're definitely right with the `SignupGoogleFragment`, it was already happening at the same moment so I moved the both calls next to each other. 
It's hard for me to find out about the `WPMainActivity` case. I'm basically tracking the moment when we retrieve the token from the magic link and I'm not sure what steps are happening between that and `onAccountChanged` call which triggers the `trackMagicLinkSignupIfNeeded`. WDYT?

**Comment by aforcier** [comment]
👍 to the move to the `SignupGoogleFragment`.

For the `WPMainActivity` (signup via magic link) case, I'm not completely sure. Here's my take:

Where it currently is, it triggers when we extract the token from the magic link. This matches the criterion of "once we've obtained the user's bearer token" for dispatching the tracks event, but at this point we wouldn't know if the token was invalid for any reason.

If we move it to where `trackCreatedAccount` is called (inside `trackMagicLinkSignupIfNeeded()`), we're relying on these things having gone well:
1. Token is registered with FluxC properly
2. `AccountAction.FETCH_ACCOUNT` was dispatched and came back from the server successfully
3. The account has been updated with a username and e-mail

My instinct would be to leave it where it is, as some of those things can go wrong and succeed later after retrying without needing a new token (so I would count the user as already 'logged in').

Since ultimately we're trying to be consistent with iOS though, @aerych could you tell us at which point this event is fired on that platform? Is it right when the token is obtained from the magic link or sometime later?

**Comment by aforcier** [comment]
Note: I'm taking this PR over for @planarvoid since he's going on vacation.

**Comment by jkmassel** [comment]
I'm bumping this to `13.1` as part of the `13.0` code freeze.

If you'd like it to be released as part of `13.0`, please merge it against `release/13.0` then DM me, and I'll be happy to issue a new beta release! :)

**Comment by aerych** [comment]
> at which point this event is fired on that platform? Is it right when the token is obtained from the magic link or sometime later?

Its actually a bit later.  (My earlier comment wasn't terribly accurate.)  The tl;dr is we bump the signed in stat once account information has been synced.  This is true for both sign up and log in flows.  

Here's the relevant code points:

- The magic links is clicked, the app is launched, and the token extracted. The user is presented with an instance of the `NUXLinkAuthViewController` to look at while [account information](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/develop/WordPressAuthenticator/NUX/NUXLinkAuthViewController.swift#L29) is synced.  The account created stat is bumped from [here](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/develop/WordPressAuthenticator/NUX/NUXLinkAuthViewController.swift#L38). Note that the stat is bumped while the api call(s) to sync account information are in flight. 

- To sync account info,`NUXLinkAuthViewController` calls a method declared in `LoginViewController` called `syncWPComAndPresentEpilogue` [#](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/develop/WordPressAuthenticator/Signin/LoginViewController.swift#L209).  This in turn calls `syncWPCom` [#](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/develop/WordPressAuthenticator/Signin/LoginViewController.swift#L229), which [calls a method](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/develop/WordPressAuthenticator/Signin/LoginViewController.swift#L234) on the authentication delegate (the actual WordPress app) to perform the sync.  When syncing account and blog information is complete the completion callback block is executed, and we see that _here_ is where we [finally bump the signed in stat](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/develop/WordPressAuthenticator/Signin/LoginViewController.swift#L238).

**Comment by aforcier** [comment]
@aerych thanks for the walkthrough on that!

Then, given that:
a. iOS is bumping the stat after account info is fetched and we want to be as consistent as possible, and
b. We're storing whether signup tracks need to be sent or not in a preference (`AppPrefs.getShouldTrackMagicLinkSignup()`), which should be triggered on the first successful account fetch even if the app is restarted,

I propose we move the call to `trackAnalyticsSignIn()` to the same place we call `trackCreatedAccount()` for the e-mail sign up flow, as @theck13 had suggested:

https://github.com/wordpress-mobile/WordPress-Android/blob/1d03591e3c0e179d3d6e6d719321425280cd1041/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java#L969-L976

WDYT @theck13 ?

**Comment by theck13** [comment]
Just for clarification, we would only move the `trackAnalyticsSignIn()` instance in `WPMainActivity` to the location in your code snippet above and leave the other `trackAnalyticsSignIn()` instances as is, right?  If so, that makes sense to me.

**Comment by aforcier** [comment]
@theck13 yep! Just [this one](https://github.com/wordpress-mobile/WordPress-Android/pull/10280/files#diff-1cf6428dcfe0d429dfdcf7398ef4c056R286) would be moved:

**Comment by aerych** [comment]
Thank you, everyone, for wrangling this!
